runtime/src/main/java/org/corfudb/runtime/Restore.java

```diff
@@ -126,12 +132,11 @@ private void restoreTable(Path filePath, UUID streamId) throws IOException {
         long startTime = System.currentTimeMillis();
 
         try (FileInputStream fileInput = new FileInputStream(filePath.toString())) {
-            TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE);
 
             // Clear table before restore
             if (restoreMode == RestoreMode.PARTIAL) {
                 SMREntry entry = new SMREntry("clear", new Array[0], Serializers.PRIMITIVE);
-                txn.logUpdate(streamId, entry);
+                nonCachedAppendSMREntries(streamId, entry);
```

Comment:
I have two concerns here:

1. What happens if the write fails? 

2. This write introduces write amplification. Entries that have been written together to multiple streams are unpacked and written individually. To hide that cost, I suggest we batch the writes.

Reply:
Thanks for reviewing. 
1.  If write fails, Restore::start throws BackupRestoreException and whoever calls restore should catch the exception and retry if necessary. Retry is simple because restore is idempotent - it clears the tables which are to be restored, and it's writing the same data anyways.
2. If it's a concern about the performance of restore - restore is carried out in special cases and is not time sensitive, so I think the extra writes should not be a big problem. Also given the complexity we have seen in checkpointer where it's hard to find a good batching strategy (batch size? data size? compressed size? serialized size?) I think it's an overkill to do that for Restore. After all after a few compaction cycles most of the data will be rewritten to checkpoint entries which is batched.

Reply:
Do you know if `BackupRestoreException` is currently being handled? Because its a `RuntimeException`, it might not have been accounted for because it simply hasn't been encountered.

Also, batching is a little easier in this context because Opaque streams are just byte arrays and you can accurately and efficiently compute the sizes. I would suggest batching because it is a little hard to predict the cost of amplification.

Even if its a rare operation, I think it would be nice for it to work as fast as possible.

Reply:
Today the Backup and Restore are invoked by python check_call() and any exception thrown from Java main function will cause check_call() to return non-zero status. That is handled by other teams.

I have added the batching to Restore.